cohttp-eio/src/body.ml

```diff
@@ -43,56 +43,78 @@ let pp_chunk fmt = function
   | Last_chunk extensions -> pp_chunk_extension fmt extensions
 
 open Reader
+open Eio.Buf_read
 
 let read_fixed t headers =
   match Http.Header.get headers "Content-length" with
   | Some v ->
       let content_length = int_of_string v in
-      let content = take_bytes content_length t in
+      let content = take content_length t in
       content
   | None -> raise @@ Invalid_argument "Request is not a fixed content body"
 
 (* Chunked encoding parser *)
 
+open Eio.Buf_read.Syntax
+
 let hex_digit = function
   | '0' .. '9' -> true
   | 'a' .. 'f' -> true
   | 'A' .. 'F' -> true
   | _ -> false
 
-let quoted_pair =
-  char '\\'
-  *> satisfy (function ' ' | '\t' | '\x21' .. '\x7E' -> true | _ -> false)
+let quoted_char =
+  let+ c = any_char in
```

Comment:
Before, it did `(qdtext <|> quoted_pair)`. The first parser reads the `\`, sees that it's not acceptable, and raises an exception, which is caught by `<|>`, which resets the state and then calls the second parser, which then reads the character again and checks that it is `\`.

Now, we just read the character once. If it's `\`, we use this parser to get the second character. Otherwise, we add the character we just read.

Reply:
Ah, yes, I was still thinking you did the translation to Buf_read parser by parser. You have already read the \ token in `quoted_string`.